pkg/operator/ceph/cluster/cephstatus.go

```diff
@@ -56,30 +56,26 @@ func newCephStatusChecker(context *clusterd.Context, clusterInfo *cephclient.Clu
 	c := &cephStatusChecker{
 		context:     context,
 		clusterInfo: clusterInfo,
-		interval:    defaultStatusCheckInterval,
+		interval:    &defaultStatusCheckInterval,
 		client:      context.Client,
 		isExternal:  clusterSpec.External.Enable,
 	}
 
 	// allow overriding the check interval with an env var on the operator
 	// Keep the existing behavior
-	var checkInterval string
+	var checkInterval *time.Duration
 	checkIntervalCRSetting := clusterSpec.HealthCheck.DaemonHealth.Status.Interval
 	checkIntervalEnv := os.Getenv("ROOK_CEPH_STATUS_CHECK_INTERVAL")
 	if checkIntervalEnv != "" {
-		checkInterval = checkIntervalEnv
-	}
-
-	if checkIntervalCRSetting != "" && checkIntervalEnv == "" {
-		checkInterval = checkIntervalCRSetting
-	}
-
-	// Set duration
-	if checkInterval != "" {
-		if duration, err := time.ParseDuration(checkInterval); err == nil {
-			logger.Infof("ceph status check interval is %s", checkInterval)
-			c.interval = duration
+		if duration, err := time.ParseDuration(checkIntervalEnv); err == nil {
+			checkInterval = &duration
 		}
```

Comment:
Why not check error != nil here?

Reply:
Because if we fail we just switch to the default, there is no error to handle.